Added all Tasks 1.2

KirillKovardakov opened the pull request.

KirillKovardakov committed:
Added all Tasks 1.2

KuzminDS commented on Task1.2/ConsoleApp1/Program.cs:
```diff
@@ -0,0 +1,139 @@
+    public static void Averages()
+    {
+            //string str = "Викентий хорошо отметил  день рождения:  покушал пиццу, посмотрел кино, пообщался со студентами в чате";
+            string str = "        Викентий    хорошо     отметил  день           рождения:  покуша,,,,,л пиццу, по...,,смотрел ки\\но, пооб.щался со сту..,дентами в чате";
```
Надо делать интерфейс, чтобы пользователь мог ввести любую строку

KuzminDS commented on Task1.2/ConsoleApp1/Program.cs:
```diff
@@ -0,0 +1,139 @@
+            {
+                sumLetters++;
+            }
+            if (char.IsSeparator(str[i]) && i < str.Length - 1 && sumLetters != 0)
```
IsSeparator пишет true даже если встречает дефис, иногда одно слово состоит из 2х частей разделенных дефисом. Ну для задания сойдет, просто в следующий раз почитай как работает метод перед тем, как его использовать.

KuzminDS commented on Task1.2/ConsoleApp1/Program.cs:
```diff
@@ -0,0 +1,139 @@
+        int counter = 0;
+        foreach (var sub in str)
+        {
+            if (sub.Length != 0)
```
Чтобы не делать эту проверку, нужно в Split указать параметр SplitOption.RemoveEmptyEntries что-то типа этого

KuzminDS commented on Task1.2/ConsoleApp1/Program.cs:
```diff
@@ -0,0 +1,139 @@
+        }
+        Console.WriteLine($"Number of words starting with a lowercase letter: {counter}");
+    }
+    public static void LowerCase2()//Второй вариант решения задачи... Я просто сначала этим вариантом решил,
```
Я думаю способ выше будет лучше

KuzminDS commented on Task1.2/ConsoleApp1/Program.cs:
```diff
@@ -0,0 +1,139 @@
+        }
+        Console.WriteLine($"Number of words starting with a lowercase letter: {counter}");
+    }
+    public static void Validator()
```
Здесь это также можно сделать через Split буквально в 5 строк

KirillKovardakov committed:
Update task1.2/1

Added line input
Added StringSplitOption.RemoveEmptyEntries parameter
Redesigned Validator

KirillKovardakov committed:
Update Validator with Split()

modified the method by removing unnecessary variables

KuzminDS commented on Task1.2/ConsoleApp1/Program.cs:
```diff
@@ -1,139 +1,151 @@
-            {
-                str[i] = char.ToUpper(str[i]);
-            }
+            // StringBuilder str = new StringBuilder
```
Нужно убрать закомменченный код

KirillKovardakov committed:
removed the commented code
